CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+* [Syntax](#syntax)
+* [Naming](#naming)
+* [Comments](#comments)
+  * [Comment Annotations](#comment-annotations)
+* [Classes](#classes--modules)
+* [Exceptions](#exceptions)
+* [Collections](#collections)
+* [Strings](#strings)
+* [Regular Expressions](#regular-expressions)
+* [Percent Literals](#percent-literals)
+* [Metaprogramming](#metaprogramming)
+
+## Source Code Layout
+
+  * Use `UTF-8` as the source file encoding. If you're using a ruby version older than 2.0, adding ```# encoding: utf-8``` on top of a ruby file will tell the compiler to use this encoding by default. If using a ruby VM older than 2.0, `UTF-8` is the default encoding.
+
+ * Use two **spaces** per indentation level (aka soft tabs). No hard tabs.
+
+  ```Ruby
+  # bad - four spaces
+  def some_method
+      do_something
+  end
+
+  # good
+  def some_method
+    do_something
+  end
+  ```
+
+ *  Don't use `;` to separate statements and expressions. As a corollary - use one expression per line.
+
+  ```Ruby
+  # bad
+  puts 'foobar'; # superfluous semicolon
+
+  puts 'foo'; puts 'bar' # two expressions on the same line
+
+  # good
+  puts 'foobar'
+
+  puts 'foo'
+  puts 'bar'
+
+  puts 'foo', 'bar' # this applies to puts in particular
+  ```
+
+  * Prefer a single-line format for class definitions with no body.
+
+  ```Ruby
+  # bad
+  class FooError < StandardError
+  end
+
+  # okish
+  class FooError < StandardError; end
+
+  # good
+  FooError = Class.new(StandardError)
+  ```
+
+  * Avoid single-line methods. Although they are somewhat popular in the wild, there are a few peculiarities about their definition syntax that make their use undesirable. At any rate - there should be no more than one expression in a single-line method.
+
+  ```Ruby
+  # bad
+  def too_much; something; something_else; end
+
+  # okish - notice that the first ; is required
+  def no_braces_method; body end
+
+  # okish - notice that the second ; is optional
+  def no_braces_method; body; end
+
+  # okish - valid syntax, but no ; makes it kind of hard to read
+  def some_method() body end
+
+  # good
+  def some_method
+    body
+  end
+  ```
+
+  One exception to the rule are empty-body methods.
+
+  ```Ruby
+  # good
+  def no_op; end
+  ```
+
+  * Use spaces around operators, after commas, colons and semicolons, around `{` and before `}`. Whitespace might be (mostly) irrelevant to the Ruby interpreter, but its proper use is the key to writing easily readable code.
+
+  ```Ruby
+  sum = 1 + 2
+  a, b = 1, 2
+  [1, 2, 3].each { |e| puts e }
+  class FooError < StandardError; end
+  ```
+
+  The only exception, regarding operators, is the exponent operator:
+
+  ```Ruby
+  # bad
+  e = M * c ** 2
+
+  # good
+  e = M * c**2
+  ```
+
+  `{` and `}` deserve a bit of clarification, since they are used for block and hash literals, as well as string interpolation. For hash literals two styles are considered acceptable.
+
+  ```Ruby
+  # good - space after { and before }
+  { one: 1, two: 2 }
```

Comment:
It's notable that because of my own style practices, Logstash uses hashrocket `{ key => value }` everywhere, not the newer Ruby 1.9+ hash-colon `{ key: value }` syntax.

We _can_ move to the new hash-colon syntax (though it has some caveats such as requiring symbols for all keys) if we determine it valuable, but if we don't, we shouldn't use hash-colon syntax in the style guide.